test/manual/atlas_connectivity.test.ts

```diff
@@ -0,0 +1,67 @@
+import { LEGACY_HELLO_COMMAND, MongoClient } from '../mongodb';
+
+/**
+ * ATLAS_CONNECTIVITY env variable is JSON
+ * Here's some typescript describing the shape:
+ *
+ * ```typescript
+ * interface AtlasConnectivity {
+ *  [atlasDeployment: string]: [normalUri: string, srvUri: string]
+ * }
+ * ```
+ *
+ * It should be an object with descriptive strings about the deployment type and version (i.e. sharded_cluster_3_4)
+ * that map to a two string tuple that are the normal URI and SRV URI, order doesn't matter, but it should be that order.
+ */
+
+describe('Atlas Connectivity', function () {
+  if (process.env.ATLAS_CONNECTIVITY == null) {
+    console.error(
+      'skipping atlas connectivity tests, ATLAS_CONNECTIVITY environment variable is not defined'
+    );
+
+    return;
+  }
+
+  const CONFIGS: Record<string, [normalUri: string, srvUri: string]> = JSON.parse(
+    process.env.ATLAS_CONNECTIVITY
+  );
+
+  let client;
+
+  afterEach(async function () {
+    await client.close();
+  });
+
+  for (const configName of Object.keys(CONFIGS)) {
+    context(configName, function () {
+      for (const connectionString of CONFIGS[configName]) {
+        const name = connectionString.includes('mongodb+srv') ? 'mongodb+srv' : 'normal';
+
+        beforeEach(function () {
+          if (configName === 'replica_set_4_4_free') {
+            const today = new Date();
+            // Making this April 1st so it is a monday
+            const april1st2024 = new Date('2024-04-01');
+            if (today < april1st2024) {
```

Comment:
This is kinda fun TBH, if the upgrade doesn't happen we'll go red again otherwise this will just become dead code and we'll start testing it again 😎

Reply:
😎